src/services/message-manager/message-manager.service.ts

```diff
@@ -1,88 +1,131 @@
-      MessageManagerService._instance = new MessageManagerService()
+    if (_.isNil(MessageManagerService._instance)) {
+      MessageManagerService._instance = new MessageManagerService();
     }
-    return MessageManagerService._instance
+    return MessageManagerService._instance;
   }
 
   public messageEvent(client: Client): void {
-    client.on('message', (msg: Message) => {
-      this._manageMessage(msg)
-    })
+    client.on(`message`, (msg: Message): void => {
+      this._manageMessage(msg);
+    });
   }
 
   private _manageMessage(message: Message): void {
     if (message.guild) {
-      if (message.author.bot) return
+      if (message.author.bot) {
+        return;
+      }
 
-      const prefix: string = PrefixManagerService.getInstance().getReadablePrefix(message.content)
+      const prefix: string = PrefixManagerService.getInstance().getReadablePrefix(
+        message.content
+      );
 
       if (message.content.startsWith(prefix)) {
-        message.channel.startTyping(1)
-
-        const args = ArgumentsManagerService.getInstance().getArguments(message, prefix)
-        const command = ArgumentsManagerService.getInstance().extractCommand(args)
+        message.channel.startTyping(1).catch((_err: string): void => {
+          console.log(
+            `The bot couldn't write in the channel: ${message.channel.id}`
+          );
+        });
+
+        const args = ArgumentsManagerService.getInstance().getArguments(
+          message,
+          prefix
+        );
+        const command = ArgumentsManagerService.getInstance().extractCommand(
+          args
+        );
 
         if (command === CommandsEnum.CHARACTER) {
-          const createArg = args.find(arg => arg === SubcommandsEnum.CHARACTER_CREATION)
-
-          if (createArg) CharacterCreationService.getInstance().init(message)
-          else CharacterCommandService.getInstance().message(message)
+          const createArg = args.find(
+            (arg): boolean => arg === SubcommandsEnum.CHARACTER_CREATION
+          );
+
+          if (createArg) {
+            CharacterCreationService.getInstance().init(message);
+          } else {
+            CharacterCommandService.getInstance().message(message);
+          }
+        } else {
+          DisplayMessageService.getInstance().message(message, `:smiling_imp:`);
         }
-        else DisplayMessageService.getInstance().message(message, ':smiling_imp:')
 
-        return message.channel.stopTyping(true)
+        return message.channel.stopTyping(true);
       }
-      else {
-        StateManagerService.getInstance().getBotState(message.author.id)
-        .then((stateFound): void => {
-          if (message.content.startsWith('exit') && stateFound.state.name !== StateNamesEnum.NORMAL) {
-            message.channel.startTyping(1)
+
+      StateManagerService.getInstance()
+        .getBotState(message.author.id)
+        .then((_foundstate): void => {
```

Comment:
So this is not the first time I saw you add a leading _.
When you do that it should only be in one case:
When a required parameter is not used but a following parameter is.
The linter will then tell you that the parameter is not used and then you will explicitly tell it "nah that's fine I know what am I doing and ignore that" by adding this leading _.
So please, search in your whole project for all the (_ variables and just remove them if really useless instead of having dead code.

Reply:
Oh... I thought it was a thing to be sure the variable was scoped to the function. Like distinguishing parent "character" from this function "_character" or something like that. I will remove them all.